dev-tools/mage/dockerbuilder.go

```diff
@@ -182,13 +182,26 @@ func (b *dockerBuilder) expandDockerfile(templatesDir string, data map[string]in
 
 func (b *dockerBuilder) dockerBuild() (string, error) {
 	tag := fmt.Sprintf("%s:%s", b.imageName, b.Version)
+	aliasTag := ""
 	if b.Snapshot {
 		tag = tag + "-SNAPSHOT"
+		// remove third number in version
+		aliasTag = fmt.Sprintf("%s:%s-SNAPSHOT", b.imageName, b.Version[0:strings.LastIndex(b.Version, ".")])
 	}
 	if repository, _ := b.ExtraVars["repository"]; repository != "" {
 		tag = fmt.Sprintf("%s/%s", repository, tag)
+		aliasTag = fmt.Sprintf("%s/%s", repository, aliasTag)
```

Comment:
`aliasTag` is empty here if `b.Snapshot` is `false`, what will add an incorrect alias if repository is set, maybe we should check here if the alias tag is empty, or if `b.Snapshot` is `true`.
```suggestion
		if b.Snapshot {
				aliasTag = fmt.Sprintf("%s/%s", repository, aliasTag)
		}
```

Reply:
I reverted the commits affecting the Go code